test/partition-test/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+cmake_minimum_required(VERSION 3.18)
+project(tests)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/../../cmake" ${CMAKE_MODULE_PATH})
+
+find_package(MPI REQUIRED)
+find_package(nlohmann_json REQUIRED)
```

Comment:
Would it be much effort to switch to yaml-cpp? This is yet another dependency for people to install. (You can ignore that for now, the tests aren't too important I guess)

Reply:
Probably it would be a tiny bit of work. But it's only for the tests after all, so it shouldn't be to worrisome.